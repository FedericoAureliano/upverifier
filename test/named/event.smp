data tResponseStatus := SUCCESS | ERROR

event eRequest := {request_id: Int}
event eResponse := {response_id: Int, status: tResponseStatus}

(* The above just means
data event := 
  | eRequest {request_id: Int} 
  | eResponse {response_id: Int, status: tResponseStatus}
*) 

function e : event_list[Int] := send(1, 2, eResponse(1, SUCCESS), empty)

print(e)