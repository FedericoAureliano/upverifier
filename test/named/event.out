send(1, 2, eResponse(1, SUCCESS), empty)
